gcc-4.8/post-link.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+if [ "$(uname)" != "Darwin" ]; then
```

Comment:
This entire PR is really for Linux.  The things that this PR fixes ("fixed" std headers, inconsistent std header locations, and inconsistent locations for `crt0.o`, etc.) aren't issues on Mac.  If OSX had multiple "distributions" like Linux, that would be a different story.  It looks like we can just skip all this stuff on Mac.

>The part below fi statement is kind of needed.

The part below the last `fi` merely verifies that "Hello World" compiles.  Generally, it should always pass -- unless the user didn't install the xcode command-line tools on her machine.  (Otherwise gcc can't find the standard library headers.)